Statistics/StatsDataModel.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Statistics
+{
+    public class StatsDataModel

```

Comment:
Why separate the computation of statistics and the statistics themselves? Any addition of statistics will require change in separate classes. If the classes always evolve together, better keep them together. This is encapsulation.